src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/Resource.sql

```diff
@@ -1,52 +1,48 @@
-﻿CREATE TABLE dbo.Resource
+﻿CREATE TABLE dbo.ResourceCurrent
 (
-    ResourceTypeId              smallint                NOT NULL,
-    ResourceId                  varchar(64)             COLLATE Latin1_General_100_CS_AS NOT NULL,
-    Version                     int                     NOT NULL,
-    IsHistory                   bit                     NOT NULL,
-    ResourceSurrogateId         bigint                  NOT NULL,
-    IsDeleted                   bit                     NOT NULL,
-    RequestMethod               varchar(10)             NULL,
-    RawResource                 varbinary(max)          NOT NULL,
-    IsRawResourceMetaSet        bit                     NOT NULL DEFAULT 0,
-    SearchParamHash             varchar(64)             NULL,
-    TransactionId               bigint                  NULL,     -- used for main CRUD operation 
-    HistoryTransactionId        bigint                  NULL      -- used by CRUD operation that moved resource version in invisible state 
+    ResourceTypeId              smallint                NOT NULL
+   ,ResourceSurrogateId         bigint                  NOT NULL
+   ,ResourceId                  varchar(64)             COLLATE Latin1_General_100_CS_AS NOT NULL
+   ,Version                     int                     NOT NULL
+   ,IsHistory                   bit                     NOT NULL CONSTRAINT DF_ResourceCurrent_IsHistory DEFAULT 0, CONSTRAINT CH_ResourceCurrent_IsHistory CHECK (IsHistory = 0)
```

Comment:
Why is the IsHistory flag still needed if historical versions will be in a different table?

Reply:
I need a column with check constraint which SQL will use to eliminate unnecessary tables when it evaluates queries with IsHistory=0 in WHERE.